lib/decode.go

```diff
@@ -64,11 +65,28 @@ func (e *DataReconstructionError) Error() string {
 }
 
 // DecodeSnapshot restores an entire snapshot to dst
-func DecodeSnapshot(repository Repository, snapshot *Snapshot, dst string) (prog chan Progress, err error) {
+func DecodeSnapshot(repository Repository, snapshot *Snapshot, dst string, excludes []string) (prog chan Progress, err error) {
 	prog = make(chan Progress)
 	go func() {
 		for _, arc := range snapshot.Archives {
 			path := filepath.Join(dst, arc.Path)
+
+			match := false
+			for _, exclude := range excludes {
+				match, err = filepath.Match(strings.ToLower(exclude), strings.ToLower(arc.Path))
+				if err != nil {
+					fmt.Println("Invalid exclude filter:", exclude)
+					return
+				}
+				if !match {
```

Comment:
This logic looks to be reversed: it would abort checking for excludes if the first exclude didn't match, ignoring further supplied excludes.

We can abort checking excludes as soon as we encounter the first match, however.